main.tf

```diff
@@ -78,22 +81,23 @@ resource "spacelift_aws_role" "this" {
   depends_on = [
     spacelift_stack.this
   ]
-  stack_id = spacelift_stack.this.id
-  role_arn = var.create_iam_role ? aws_iam_role.this[0].arn : var.execution_role_arn
+  external_id = var.create_iam_role ? local.external_id : var.execution_role_external_id
```

Comment:
External IDs are only valid when `generate_credentials_in_worker` is true, so I think what might make more sense is something like the following:

- Add a `aws_custom_external_id` variable to contain the external ID.
- Update your `external_id` local to either use `{account}@{stack}` like you currently do, or to use `var.aws_custom_external_id`.
- If `aws_custom_external_id` is set, you should also set `generate_credentials_in_worker` to `true`, and pass it to `external_id`.
- If `aws_custom_external_id` isn't set, you can just leave `external_id` empty - in that case we ignore the value passed to the resource and generate it automatically when assuming the role.